server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -230,23 +234,25 @@ private synchronized void syncSegments(boolean isRetry) {
                         // Create a map of file name to size and update the refresh segment tracker
                         updateLocalSizeMapAndTracker(localSegmentsPostRefresh);
 
-                        // Start the segments files upload
-                        boolean newSegmentsUploadStatus = uploadNewSegments(localSegmentsPostRefresh);
-                        if (newSegmentsUploadStatus) {
-                            segmentInfoSnapshotFilename = uploadSegmentInfosSnapshot(latestSegmentInfos.get(), segmentInfos);
-                            localSegmentsPostRefresh.add(segmentInfoSnapshotFilename);
-
-                            // Start metadata file upload
-                            remoteDirectory.uploadMetadata(
-                                localSegmentsPostRefresh,
-                                storeDirectory,
-                                indexShard.getOperationPrimaryTerm(),
-                                segmentInfos.getGeneration()
-                            );
-                            clearStaleFilesFromLocalSegmentChecksumMap(localSegmentsPostRefresh);
-                            onSuccessfulSegmentsSync(refreshTimeMs, refreshSeqNo);
+                        List<String> segmentFilesToUpload = localSegmentsPostRefresh.stream()
+                            .filter(f -> skipUpload(f) == false)
+                            .collect(Collectors.toList());
+
+                        if (segmentFilesToUpload.isEmpty() || uploadNewSegments(segmentFilesToUpload)) {
```

Comment:
If `uploadNewSegments` returns an enum where one of the values denote skipped downloading of all the segments, can it be used along with `isRetry`?

Reply:
I had thought of this. If we have this kind of status, we would need to define the overall status too because that would be used to update the segment tracker on the final status. Let me discuss this more with you.